packages/itwinui-react/src/core/Stepper/Stepper.tsx

```diff
@@ -41,6 +41,10 @@ export type StepperProps = {
    *  Option to provide localized strings.
    */
   localization?: StepperLocalization;
+  /**
+   * Custom content passed for completed steps.
+   */
+  stepCircleRenderer?: (completedIndex: number) => React.ReactNode;
```

Comment:
This API feels a bit strange. Is there a reason we can't allow passing a ReactNode in the steps array itself?

e.g.:

```jsx
const steps = [
  { name: 'First Step', stepContent: () => <SvgCheckmarkSmall /> },
  { name: 'Second Step', stepContent: () => <SvgCheckmarkSmall /> },
  { name: 'Third Step', stepContent: () => <SvgCheckmarkSmall /> },
  { name: 'Last Step', stepContent: () => <SvgCheckmarkSmall /> },
];
```

Reply:
`Select` and especially `ComboBox` a bit different because they often require large datasets, and we fully control the rendering of options (and pass certain information back to the consumer). Still, I'm not happy with those item renderer APIs; I think I would design it differently if I was to do it again.

This `Stepper` case is a lot simpler though, so it makes sense to just ask for it within the steps. In some ways it's similar to handling of icons within `Select`/`ComboBox` options.